src/Common/Exception.cpp

```diff
@@ -781,4 +781,27 @@ ExecutionStatus ExecutionStatus::fromText(const std::string & data)
     return status;
 }
 
+std::exception_ptr copyMutableException(std::exception_ptr ptr)
+{
+    try
+    {
+        std::rethrow_exception(ptr);
+    }
+    catch (Poco::Exception & e)
+    {
+        try
+        {
+            e.rethrow();
+        }
+        catch (...)
+        {
+            return std::current_exception();
+        }
```

Comment:
This catches the exception, then the call site immediately rethrows it. Change the function to something like `rethrowExceptionCopy(std::exception_ptr)` to avoid the extra roundtrip?